pkg/peertls/tlsopts/options_test.go

```diff
@@ -103,7 +106,48 @@ func TestNewOptions(t *testing.T) {
 		assert.NoError(t, err)
 		assert.True(t, reflect.DeepEqual(fi, opts.Ident))
 		assert.Equal(t, c.config, opts.Config)
-		assert.Len(t, opts.VerificationFuncs, c.pcvFuncsLen)
+		assert.Len(t, opts.VerificationFuncs.Client(), c.clientVerificationFuncsLen)
+		assert.Len(t, opts.VerificationFuncs.Server(), c.serverVerificationFuncsLen)
+	}
+}
+
+type identFunc func(int) (*identity.FullIdentity, error)
+
+func TestOptions_ServerOption_Peer_CA_Whitelist(t *testing.T) {
+	ctx := testcontext.New(t)
+
+	planet, err := testplanet.New(t, 0, 2, 0)
+	require.NoError(t, err)
+
+	planet.Start(ctx)
+	defer ctx.Check(planet.Shutdown)
+
+	target := planet.StorageNodes[1].Local()
+
+	testCases := []struct {
+		name   string
+		identF identFunc
+	}{
+		{"unsigned client identity", testplanet.PregeneratedIdentity},
+		{"signed client identity", testplanet.PregeneratedSignedIdentity},
+	}
+	for _, testCase := range testCases {
+		t.Run(testCase.name, func(t *testing.T) {
+			ident, err := testCase.identF(0)
```

Comment:
maybe doesn't need to be tested here in this test necessarily, but should `ident` also be checked to make sure it has appropriate fields set?